### ISSUE TITLE
Support data.world as a datasource

### ISSUE THREAD

**Issue by rflprr** [plotly/falcon-sql-client#186]
data.world would be a great general purpose, low entry barrier datasource for plot.ly.
It should be a very straightforward integration based on the implementation of existing data.world connectors for Tableau (https://github.com/datadotworld/tableau-connector) and Google Data Studio (https://github.com/datadotworld/data-studio-connector).

**Comment by bpostlethwaite** [comment]
That is a great idea, thanks for the feedback!

**Comment by n-riesco** [comment]
@rflprr I'm labelling this issue as `community`. I also think this is a good first contribution. If you feel like contributing, I can point you to the files that need changing and help through the process.

**Comment by rflprr** [comment]
Hi, @n-riesco. We would like to contribute and could start as early as this week. What's the best way for us to get started?

**Comment by n-riesco** [comment]
@rflprr 

First of all, you need to set up your dev machine:
- install Node.js v6
- install yarn v1
- clone Falcon's repository, build and run:
  ```
  $ git clone https://github.com/plotly/falcon-sql-client.git
  $ cd falcon-sql-client
  $ yarn install
  $ yarn run rebuild:modules:electron
  $ yarn run build
  $ yarn run start-hot
  ```
Once the dev machine is setup. Here's a list of files you need to look at to add a new connector:
- `backend/persistent/datastores/Datastores.js`
- `app/constants/constants.js`
- `app/images`
- `app/actions/sessions.js`
- `app/components/Settings/Tabs/Tab.react.js`
- `app/components/Settings/Settings.react.js`

As an example of what needs doing, you could also have a look at the following commits from the PR implementing the Impala connector:
- UI changes: 2d1477f2435aae3b37ef1a27ec63daa6100d6eb1
- Backend implementation: 67ae51d3739a3f147b49e1ad1720ba326a5829b6
- Add a sample connection: 9e85a2bab3aed6ce3c6267bdc99037476dd0a997
- Add tests: https://github.com/plotly/falcon-sql-client/blob/master/test/backend/datastores.ibmdb.spec.js

Hopefully, I didn't miss anything ;), but if I did or you need more help, don't hesitate to contact me.

**Comment by kndungu** [comment]
@n-riesco 

I've just made a PR for the initial implementation.

Pending: 
<img width="332" alt="screen shot 2018-01-02 at 18 18 19" src="https://user-images.githubusercontent.com/17295379/34488384-61a82ab8-efe9-11e7-816a-d0cca6969e7c.png">

I've as yet been unable to make the database name appear on the side panel's tree view, might you have some thoughts on this?

Token:
Connection to a dataset requires the user to provide a valid data.world API token. Due to various reasons we'd prefer not to provide a publicly accessible token. This is the reason as to why tests are not part of the PR. Your thoughts on this would also be appreciated 

cc @rflprr

**Comment by n-riesco** [comment]
Closed via #332
